codec-http/src/main/java/io/netty/handler/codec/http/HttpKeepAliveHandler.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.handler.codec.http;
+
+import io.netty.channel.ChannelDuplexHandler;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelFutureListener;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.ChannelPromise;
+import io.netty.util.Attribute;
+import io.netty.util.AttributeKey;
+
+import static io.netty.handler.codec.http.HttpUtil.*;
+import static java.lang.Boolean.*;
+
+/**
+ * HttpKeepAliveHandler helps maintain persistent connections when requested.
+ * <p>
+ * When a persistent connection request arrives then HttpKeepAliveHandler will make sure that the response also contains
+ * the proper 'Connection' header.  The channel will automatically be closed for any LastHttpContent being written
+ * through this handler, iff the connection cannot be persisted.
+ * <p>
+ * Since {@link HttpKeepAliveHandler} expects {@link HttpObject}s it should be added after {@link HttpServerCodec} but
+ * before any other handlers that might send a {@link HttpResponse}.
+ * <blockquote>
+ * <pre>
+ *  {@link ChannelPipeline} p = ...;
+ *  ...
+ *  p.addLast("serverCodec", new {@link HttpServerCodec}());
+ *  p.addLast("httpKeepAlive", <b>new {@link HttpKeepAliveHandler}()</b>);
+ *  p.addLast("aggregator", new {@link HttpObjectAggregator}(1048576));
+ *  ...
+ *  p.addLast("handler", new HttpRequestHandler());
+ *  </pre>
+ * </blockquote>
+ */
+@ChannelHandler.Sharable
+public class HttpKeepAliveHandler extends ChannelDuplexHandler {
+
+    private static final String MULTIPART_PREFIX = "multipart";
+
+    /**
+     * Will close the channel if the connection couldn't be persisted.
+     *
+     * Only works if HttpKeepAliveHandler is part of the pipeline.
+     */
+    public static final ChannelFutureListener CLOSE_UNLESS_KEEP_ALIVE = new ChannelFutureListener() {
+        @SuppressWarnings("unused")
```

Comment:
I finally got the inspections working and this was flagged as unused.  I took the SuppressWarnings off and reran the inspection, but it isn't getting flagged this time, which makes sense since I'm clearly using it.  My guess is there was some kind of issue with my project setup.  Removing it now.